File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/aonsolutions/Note.java
package com.esferalia.aon.occam.api.model.aonsolutions;

import java.io.Serializable;
import java.util.Date;

public class Note implements Serializable {
	
	private static final long serialVersionUID = -6673378118872331822L;
	
	public Note() {}
	
	private Integer id;
	private Integer domain;
	private String subject;
	private Date date;
	private Integer owner;
	private String note;
	private boolean archive = false;
	private boolean pinpUp = false;
	private String noteTag;
	private String color;
	private Date archiveDate;
	private Date modificationDate;
	private Date creationDate;
	
	public Integer getId() {
		return id;
	}
	public Note setId(Integer id) {
		this.id = id;
		return this;
	}
	public Integer getDomain() {
		return domain;
	}
	public Note setDomain(Integer domain) {
		this.domain = domain;
		return this;
	}
	public String getSubject() {
		return subject;
	}
	public Note setSubject(String subject) {
		this.subject = subject;
		return this;
	}
	public Date getDate() {
		return date;
	}
	public Note setDate(Date date) {
		this.date = date;
		return this;
	}
	public Integer getOwner() {
		return owner;
	}
	public Note setOwner(Integer owner) {
		this.owner = owner;
		return this;
	}
	public String getNote() {
		return note;
	}
	public Note setNote(String note) {
		this.note = note;
		return this;
	}
	public boolean isArchive() {
		return archive;
	}
	public Note setArchive(boolean archive) {
		this.archive = archive;
		return this;
	}
	public boolean isPinpUp() {
		return pinpUp;
	}
	public Note setPinpUp(boolean pinpUp) {
		this.pinpUp = pinpUp;
		return this;
	}
	public String getNoteTag() {
		return noteTag;
	}
	public Note setNoteTag(String noteTag) {
		this.noteTag = noteTag;
		return this;
	}
	public String getColor() {
		return color;
	}
	public Note setColor(String color) {
		this.color = color;
		return this;
	}
	public Date getArchiveDate() {
		return archiveDate;
	}
	public Note setArchiveDate(Date archiveDate) {
		this.archiveDate = archiveDate;
		return this;
	}
	public Date getModificationDate() {
		return modificationDate;
	}
	public Note setModificationDate(Date modificationDate) {
		this.modificationDate = modificationDate;
		return this;
	}
	public Date getCreationDate() {
		return creationDate;
	}
	public Note setCreationDate(Date creationDate) {
		this.creationDate = creationDate;
		return this;
	}
	
}

File: aon-solutions/src/main/webapp/models/note/Note.js
import { AonDateUtils } from "../../modules/utils/AonDateUtils.js";
import { Domain } from "../Domain.js";

export class Note {
  id;
  domain;
  subject;
  date;
  owner;
  note;
  archive;
  pinUp;
  noteTag;
  color;
  archiveDate;
  creationDate;
  modificationDate;

  constructor(note) {
    if(note) {
      this.id               = note.id || undefined;
      this.domain           = note.domain || new Domain().getId();
      this.subject          = note.subject || "";
      this.note             = note.note || "";
      this.owner            = note.owner || undefined;
      this.date             = note.date && !note.date.includes("0001-01-01") ? AonDateUtils.formatDateOrigin(note.date) : null;
      this.archive          = note.archive || false;
      this.pinUp            = note.pinUp || false;
      this.noteTag          = note.noteTag || "";
      this.color            = note.color || "";
      this.archiveDate      = note.archiveDate && !note.archiveDate.includes("0001-01-01") ? AonDateUtils.formatDateOrigin(note.archiveDate) : null;
      this.creationDate     = note.creationDate && !note.creationDate.includes("0001-01-01") ? AonDateUtils.formatDateOrigin(note.creationDate) : null;
      this.modificationDate = note.modificationDate && !note.modificationDate.includes("0001-01-01") ? AonDateUtils.formatDateOrigin(note.modificationDate) : null;
    } else {
      this.id               = undefined;
      this.domain           = new Domain().getId(); 
      this.subject          = "";
      this.note             = undefined;
      this.owner            = undefined;
      this.date             = null;
      this.archive          = false;
      this.pinUp            = false;
      this.noteTag          = undefined;
      this.color            = undefined;
      this.archiveDate      = null;
      this.creationDate     = null;
      this.modificationDate = null;
    }
  }

  getId() {
    return this.id;
  }

  setId(id) {
    this.id = id;
  }

  getSubject() {
    return this.subject;
  }

  setSubject(subject) {
    this.subject = subject;
  }

  getDate() {
    return this.date;
  }

  setDate(date) {
    this.date = date;
  }

  getOwner() {
    return this.owner;
  }

  setOwner(owner) {
    this.owner = owner;
  }

  getNote() {
    return this.note;
  }

  setNote(note) {
    this.note = note;
  }

  getDomain() {
    return this.domain;
  }

  setDomain(domain) {
    this.domain = domain;
  }

  getArchive() {
    return this.archive;
  }

  setArchive(archive) {
    this.archive = archive;
  }

  getPinUp() {
    return this.pinUp;
  }

  setPinUp(pinUp) {
    this.pinUp = pinUp;
  }

  getNoteTag() {
    return this.noteTag;
  }

  setNoteTag(noteTag) {
    this.noteTag = noteTag;
  }

  getColor() {
    return this.color;
  }

  setColor(color) {
    this.color = color;
  }

  getArchiveDate() {
    return this.archiveDate;
  }

  setArchiveDate(archiveDate) {
    this.archiveDate = archiveDate;
  }

  getModificationDate() {
    return this.modificationDate;
  }

  setModificationDate(modificationDate) {
    this.modificationDate = modificationDate;
  }

  getCreationDate() {
    return this.creationDate;
  }

  setCreationDate(creationDate) {
    this.creationDate = creationDate;
  }

}

